homeassistant/components/lcn/websocket.py

```diff
@@ -0,0 +1,420 @@
+"""LCN Websocket API."""
+
+from __future__ import annotations
+
+from copy import deepcopy
+from typing import Final
+
+import lcn_frontend as lcn_panel
+import voluptuous as vol
+
+from homeassistant.components import panel_custom, websocket_api
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_ADDRESS,
+    CONF_DEVICES,
+    CONF_DOMAIN,
+    CONF_ENTITIES,
+    CONF_ID,
+    CONF_NAME,
+    CONF_RESOURCE,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr, entity_registry as er
+import homeassistant.helpers.config_validation as cv
+
+from .const import (
+    ADD_ENTITIES_CALLBACKS,
+    CONF_DOMAIN_DATA,
+    CONF_HARDWARE_SERIAL,
+    CONF_HARDWARE_TYPE,
+    CONF_SOFTWARE_SERIAL,
+    CONNECTION,
+    DOMAIN,
+)
+from .helpers import (
+    DeviceConnectionType,
+    async_update_device_config,
+    generate_unique_id,
+    get_device_config,
+    get_device_connection,
+    get_resource,
+    purge_device_registry,
+    purge_entity_registry,
+    register_lcn_address_devices,
+)
+from .schemas import (
+    ADDRESS_SCHEMA,
+    DOMAIN_DATA_BINARY_SENSOR,
+    DOMAIN_DATA_CLIMATE,
+    DOMAIN_DATA_COVER,
+    DOMAIN_DATA_LIGHT,
+    DOMAIN_DATA_SCENE,
+    DOMAIN_DATA_SENSOR,
+    DOMAIN_DATA_SWITCH,
+)
+
+URL_BASE: Final = "/lcn_static"
+
+
+async def register_panel(hass: HomeAssistant) -> None:
+    """Register the LCN Panel and Websocket API."""
+    websocket_api.async_register_command(hass, websocket_get_hosts)
+    websocket_api.async_register_command(hass, websocket_get_device_configs)
+    websocket_api.async_register_command(hass, websocket_get_entity_configs)
+    websocket_api.async_register_command(hass, websocket_scan_devices)
+    websocket_api.async_register_command(hass, websocket_add_device)
+    websocket_api.async_register_command(hass, websocket_delete_device)
+    websocket_api.async_register_command(hass, websocket_add_entity)
+    websocket_api.async_register_command(hass, websocket_delete_entity)
+
+    if DOMAIN not in hass.data.get("frontend_panels", {}):
+        hass.http.register_static_path(
+            URL_BASE,
+            path=lcn_panel.locate_dir(),
+            cache_headers=lcn_panel.is_prod_build,
+        )
+        await panel_custom.async_register_panel(
+            hass=hass,
+            frontend_url_path=DOMAIN,
+            webcomponent_name=lcn_panel.webcomponent_name,
+            sidebar_title=DOMAIN.upper(),
+            sidebar_icon="mdi:cogs",
```

Comment:
Don't add this as a sidebar, instead add it as a configuration panel.
```suggestion
            config_panel_domain=DOMAIN,
```

Background in PR https://github.com/home-assistant/core/pull/96245

Reply:
Done